File: src/visualizer/akasha/terms.py
"""What the screen calls the store's three levels.

The API, the URLs and the payloads speak MongoDB -- ``database`` ->
``collection`` -> ``document`` -- and always will. Those names are not
decoration, they are the addressing scheme: the same triple keys every grant,
every wikilink (``[[db/col/id]]``) and every chronos ``EntityRef``. Renaming
them would be a migration with nothing to show for it.

The *screen*, though, is read by a novelist, who has a world with categories of
articles in it. This module is the seam between the two vocabularies, so
changing what the UI calls things is one edit here rather than a hunt through
every view.

``static/js/terms.js`` carries the same table for the browser, and
``tests/akasha/test_terms.py`` keeps the two from drifting apart.

Deliberately *not* applied to the admin console: it manages grants against the
API's scopes, so the API's words are the ones that make it legible.
"""

TERMS = {
    "database": {
        "one": "world",
        "many": "worlds",
        "One": "World",
        "Many": "Worlds",
    },
    "collection": {
        "one": "category",
        "many": "categories",
        "One": "Category",
        "Many": "Categories",
    },
    "document": {
        "one": "article",
        "many": "articles",
        "One": "Article",
        "Many": "Articles",
    },
}

File: src/visualizer/akasha/static/js/terms.js
// What the screen calls the store's three levels.
//
// The API, the URLs and the payloads speak MongoDB — database → collection →
// document — and always will: that triple is the addressing scheme behind every
// grant, every wikilink and every chronos reference. The screen speaks to a
// novelist instead, who has a world with categories of articles in it. Keeping
// the two apart means the UI can be renamed without touching a single route.
//
// The mirror of `terms.py`, which serves the Jinja pages; a test compares them
// so they cannot drift.

export const T = {
  database: { one: "world", many: "worlds", One: "World", Many: "Worlds" },
  collection: { one: "category", many: "categories", One: "Category", Many: "Categories" },
  document: { one: "article", many: "articles", One: "Article", Many: "Articles" },
};

// "1 article" / "4 categories" — the plural is looked up, not guessed at by
// adding an s, which is exactly where "categorys" would have come from.
export function count(n, term) {
  return `${n} ${n === 1 ? term.one : term.many}`;
}
